autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -0,0 +1,533 @@
+            self.predict_function = self._predict_regression
+        else:
+            if not isinstance(self.configuration, Configuration):
+                self.model_class = MyDummyClassifier
+            else:
+                if self.task_type in TABULAR_TASKS:
+                    self.model_class = TabularClassificationPipeline
+                elif self.task_type in IMAGE_TASKS:
+                    self.model_class = ImageClassificationPipeline
+                else:
+                    raise ValueError('task {} not available'.format(self.task_type))
+            self.predict_function = self._predict_regression
+        if self.task_type in TABULAR_TASKS:
+            info.update({'numerical_columns': self.datamanager.numerical_columns,
+                         'categorical_columns': self.datamanager.categorical_columns})
+        self.dataset_properties = self.datamanager.get_dataset_properties(get_dataset_requirements(info))
+
+        self._init_params = {'dataset_properties': self.dataset_properties}
+        if init_params is not None:
+            self._init_params.update(init_params)
+        self._init_params.update({
+            'X_train': self.X_train,
+            'y_train': self.y_train,
+            'X_test': self.X_test,
+            'y_test': self.y_test,
+            'backend': self.backend,
+            'logger_port': logger_port
+        })
+
+        if num_run is None:
+            num_run = 0
+        self.num_run = num_run
+
+        logger_name = '%s(%d)' % (self.__class__.__name__.split('.')[-1],
+                                  self.seed)  # TODO: Add name to dataset class
+        if logger_port is None:
+            logger_port = logging.handlers.DEFAULT_TCP_LOGGING_PORT
+        self.logger = get_named_client_logger(output_dir=self.backend.temporary_directory, name=logger_name,
+                                              port=logger_port)
+        self.logger.debug("Dataset Properties created in train_evaluator: {}".format(self.dataset_properties))
+        self.Y_optimization = None
+        self.Y_actual_train = None
+
+        self.budget = budget
+        self.budget_type = budget_type
+        default_pipeline_options = self.model_class.get_default_pipeline_options()
+        if self.budget_type is not None:
+            if self.budget_type == 'runtime':
+                default_pipeline_options['runtime'] = self.budget
+                if 'epochs' in default_pipeline_options:
+                    del default_pipeline_options['epochs']
+            elif self.budget_type == 'epochs':
+                default_pipeline_options['epochs'] = self.budget
+                if 'runtime' in default_pipeline_options:
+                    del default_pipeline_options['runtime']
+            default_pipeline_options['budget_type'] = self.budget_type
+        self.logger.debug("Default pipeline options: {}".format(default_pipeline_options))
+        self._init_params.update({**default_pipeline_options})
```

Comment:
This would override any parameters that have previously been specified if they also occur in the default options. Maybe `self._init_params = {**default_pipeline_options, **self._init_params}` would be safer